core/src/main/scala/com/microsoft/ml/spark/codegen/DotnetWrappable.scala

```diff
@@ -0,0 +1,522 @@
+      case _: TypedIntArrayParam | _: TypedDoubleArrayParam =>
+        s"""
+           |$docString
+           |public ${getParamInfo(p).dotnetType} Get$capName()
+           |{
+           |    JvmObjectReference jvmObject = (JvmObjectReference)Reference.Invoke(\"get$capName\");
+           |    return (${getParamInfo(p).dotnetType})jvmObject.Invoke(\"array\");
+           |}
+           |""".stripMargin
+      case _: UntypedArrayParam =>
+        s"""
+           |$docString
+           |public ${getParamInfo(p).dotnetType} Get$capName()
+           |{
+           |    JvmObjectReference[] jvmObjects = (JvmObjectReference[])Reference.Invoke(\"get$capName\");
+           |    object[] result = new object[jvmObjects.Length];
+           |    for (int i = 0; i < result.Length; i++)
+           |    {
+           |        result[i] = SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+           |            "com.microsoft.ml.spark.codegen.DotnetHelper", "mapScalaToJava", (object)jvmObjects[i]);
+           |    }
+           |    return result;
+           |}
+           |""".stripMargin
+      case _: ArrayMapParam =>
+        s"""
+           |$docString
+           |public ${getParamInfo(p).dotnetType} Get$capName()
+           |{
+           |    JvmObjectReference[] jvmObjects = (JvmObjectReference[])Reference.Invoke(\"get$capName\");
+           |    Dictionary<string, object>[] result = new Dictionary<string, object>[jvmObjects.Length];
+           |    JvmObjectReference hashMap;
+           |    JvmObjectReference[] keySet;
+           |    Dictionary<string, object> dic;
+           |    object val;
+           |    for (int i = 0; i < result.Length; i++)
+           |    {
+           |        hashMap = (JvmObjectReference)SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+           |            "com.microsoft.ml.spark.codegen.DotnetHelper", "convertToJavaMap", jvmObjects[i]);
+           |        keySet = (JvmObjectReference[])(
+           |            (JvmObjectReference)hashMap.Invoke("keySet")).Invoke("toArray");
+           |        dic = new Dictionary<string, object>();
+           |        foreach (var k in keySet)
+           |        {
+           |            val = SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+           |                "com.microsoft.ml.spark.codegen.DotnetHelper", "mapScalaToJava", hashMap.Invoke("get", k));
+           |            dic.Add((string)k.Invoke("toString"), val);
+           |        }
+           |        result[i] = dic;
+           |    }
+           |    return result;
+           |}
+           |""".stripMargin
+      case _: EstimatorParam | _: ModelParam | _: TransformerParam | _: EvaluatorParam | _: PipelineStageParam =>
+        val dType = p match {
+          case _: EstimatorParam | _: ModelParam =>
+            getParamInfo(p).dotnetType.substring(5, getParamInfo(p).dotnetType.length - 3) + "<object>"
+          case _ => getParamInfo(p).dotnetType
+        }
+        s"""
```

Comment:
We might want to put this info into the WrappableParam trait so that we don't need to pollute the core generator code

Reply:
That's a great point! I should add another field to capture the different return type